.github/workflows/deploy.yml

```diff
@@ -1,36 +1,43 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI
 
-# Controls when the workflow will run
 on:
   # Triggers the workflow on push or pull request events but only for the master branch
   push:
     branches: [master]
 
+  pull_request:
+    branches: [master]
+
```

Comment:
Are you sure? If I'm correct, it will deploy in production as soon as we open a PR on master

Reply:
See my comment below the other observation :)